lib/searchkick/index.rb

```diff
@@ -420,14 +420,23 @@ def full_reindex_async(scope)
       if scope.respond_to?(:primary_key)
         # TODO expire Redis key
         primary_key = scope.primary_key
-        starting_id = scope.minimum(primary_key) || 0
-        max_id = scope.maximum(primary_key) || 0
-        batches_count = ((max_id - starting_id + 1) / batch_size.to_f).ceil
-
-        batches_count.times do |i|
-          batch_id = i + 1
-          min_id = starting_id + (i * batch_size)
-          bulk_reindex_job scope, batch_id, min_id: min_id, max_id: min_id + batch_size - 1
+
+
+        if %i(integer float decimal).include?(scope.column_for_attribute(primary_key).sql_type_metadata)
```

Comment:
1. I was concerned about this as well, I'm not sure how consistent it is. It seems to be a generic API that can be over ridden by specific adapters if needed.
2. I don't think so. I thought I'd submit the PR before putting the time into investigating this. I'll do some research and get it updated.